test/integration/minimal/controls/minimal.rb

```diff
@@ -0,0 +1,80 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'json'
+
+project_name = attribute('project_name')
+region = attribute('region')
+random_suffix = attribute('random_suffix')
+gcloud_args = "--format json --project #{project_name}"
+_ = `gcloud config set project #{region}`
+_ = `gcloud auth activate-service-account --key-file ${GOOGLE_APPLICATION_CREDENTIALS}`
+
+control 'lb-internal-minimal' do
+  title 'A minimal example of running an internal load balancer.'
+
+  describe google_compute_firewall(project: project_name, name: "test-ilb-#{random_suffix}-ilb-fw") do
+    it { should allow_source_tags_only ["source-tag-foo"] }
+    it { should allow_target_tags_only ["target-tag-bar"] }
+    it { should allow_port_protocol("8080", "tcp") }
+    its('allowed_ssh?')  { should be false }
+    its('allowed_http?')  { should be false }
+    its('direction') { should eq "INGRESS" }
+    it { should_not allow_ip_ranges ["0.0.0.0/0"] }
+  end
+
+  describe google_compute_firewall(project: project_name, name: "test-ilb-#{random_suffix}-hc") do
+    it { should_not allow_port_protocol("22", "tcp") }
+    it { should allow_port_protocol("8081", "tcp") }
+    its('direction') { should eq "INGRESS" }
+    its('allowed_ssh?')  { should be false }
+    its('allowed_http?')  { should be false }
+    it { should_not allow_ip_ranges ["0.0.0.0/0"] }
+    it { should allow_ip_ranges_only ["130.211.0.0/22", "35.191.0.0/16"] }
+  end
+
+  describe google_compute_forwarding_rule(project: project_name, region: region, name: "test-ilb-#{random_suffix}") do
+    its('load_balancing_scheme') { should eq "INTERNAL" }
+    its('network') { should match "test-#{random_suffix}" }
+    its('backend_service') { should match "test-ilb-#{random_suffix}" }
+  end
+
+  health_checks = JSON.parse(`gcloud compute health-checks list --filter="name:( test-ilb-#{random_suffix}-hc )" #{gcloud_args}`)
+  describe 'The health check' do
+    it "exists and there is only 1" do
+      expect(health_checks.length).to eq(1)
+    end
+    it "is listening on 8080" do
+      expect(health_checks[0]['tcpHealthCheck']['port']).to eq(8081)
+    end
+    it "is of type TCP" do
+      expect(health_checks[0]['type']).to eq('TCP')
+    end
+  end
+
+  backend_services = JSON.parse(`gcloud compute backend-services list --filter="name:( test-ilb-#{random_suffix} )" #{gcloud_args}`)
+  puts backend_services
+  describe 'The backend service' do
+    it "exists and there is only 1" do
+      expect(backend_services.length).to eq(1)
+    end
+    it "has one health check and it's the one we created" do
+      expect(backend_services[0]['healthChecks'].length).to eq(1)
+      expect(backend_services[0]['healthChecks'][0]).to end_with "test-ilb-#{random_suffix}-hc"
```

Comment:
We should limit tests to single assertions to simplify debugging when tests fail.